hack/shellcheck-test.sh

```diff
@@ -15,28 +15,30 @@ test_shellcheck() {
                         return 0
                 fi
                 #shell check -x -e SC2181,SC2029,SC1091,SC1090,SC2012 "${1}"
-                shellcheck -x -e SC2181 "${1}"
+                "${SHELLCHECK}" -x -e SC2181 "${1}"
         else
                 return 0
         fi
 }
 
-SHELLCHECK="$(command -v shellcheck 2>/dev/null)"
-
 SCRIPT_DIR="$(cd "$(dirname "${0}")" && pwd)"
 
 BASE_DIR="$(cd "${SCRIPT_DIR}/.." && pwd)"
 
-if [[ -z "${SHELLCHECK}" ]]; then
-        echo "warning: could not find shellcheck ... installing shellcheck" >&2
-        scversion="stable"
-        FILE="${OUTDIR_TOOLS}/shellcheck"
-        wget -qO- "https://storage.googleapis.com/shellcheck/shellcheck-${scversion?}.linux.x86_64.tar.xz" | tar -xJv
-        cp -f "shellcheck-${scversion}/shellcheck" "${FILE}"
-        if [ -f "$FILE" ]; then
-                SHELLCHECK="${FILE}"
-        fi
+SC_VERSION="stable"
+echo "Installing shellcheck... (${SC_VERSION?})" >&2
+FILE="${OUTDIR_TOOLS}/shellcheck"
+if [ "$OS_TYPE" == "Darwin" ]; then
+        wget -qO- "https://github.com/koalaman/shellcheck/releases/download/${SC_VERSION?}/shellcheck-${SC_VERSION?}.darwin.x86_64.tar.xz" | tar -xJv
+else
+        wget -qO- "https://storage.googleapis.com/shellcheck/shellcheck-${SC_VERSION?}.linux.x86_64.tar.xz" | tar -xJv
 fi
+cp -f "shellcheck-${SC_VERSION}/shellcheck" "${FILE}"
```

Comment:
Don't install it every time. We want to skip this if the binary is already downloaded.

Reply:
Do you want me the search for it under the tools folder?

Reply:
I think it should be fine to just replace `if [[ -z "${SHELLCHECK}" ]]; then` with `if [ -f "${SHELLCHECK}" ]; then`. If we assume we will always call shellcheck from the same place, I think all we'd have to do is check if the file exists.

Reply:
Fixed it differently, will not download if already exists in the tools folder